### ISSUE TITLE
Generation for custom connector generates hard coded value

### ISSUE THREAD

**Issue by mikaelsand** [jeffhollan/LogicAppTemplateCreator#52]
I tried to fix this in the code, but I lack the skillz.
When creating a template with a custom connector. The ARM template contains a value for the connector. This value would benefit from being parameterized. Example:
```json
"id": "[concat('/subscriptions/',subscription().subscriptionId,'/resourceGroups/',parameters('MyValue-CustomConnector-ResourceGroup'),'/providers/Microsoft.Web/customApis/MyValue-CustomConnector-SOAPPassthru-DEV')]"
```
would be great if it looked like this:
```json
"id": "[concat('/subscriptions/',subscription().subscriptionId,'/resourceGroups/',parameters('MyValue-CustomConnector-ResourceGroup'),'/providers/Microsoft.Web/customApis/', parameters('MyValue-CustomConnector-Connectorname)]",
```

**Comment by hariharan618** [comment]
I will try to fix this

**Comment by MLogdberg** [comment]
This should be the case, we even have a test for exactly this scenario. 
You are talking about the CustomConnector connection inside the Logic App ARM Template right?

According t my test class CustomConnectorTest and it's tests it will be generated as follows:

`
"parameters": {
            "$connections": {
              "value": {
                "LDAPAdapter": {
                  "id": "[concat('/subscriptions/',subscription().subscriptionId,'/resourceGroups/',parameters('LDAPAdapter-ResourceGroup'),'/providers/Microsoft.Web/customApis/',parameters('LDAPAdapter_name'),'')]",
                  "connectionId": "[resourceId('Microsoft.Web/connections', parameters('LDAPAdapter_name'))]",
                  "connectionName": "[parameters('LDAPAdapter_name')]"
                },
                "azureblob": {
                  "id": "[concat('/subscriptions/',subscription().subscriptionId,'/providers/Microsoft.Web/locations/',parameters('logicAppLocation'),'/managedApis/azureblob')]",
                  "connectionId": "[resourceId('Microsoft.Web/connections', parameters('azureblob_name'))]",
                  "connectionName": "[parameters('azureblob_name')]"
                }
              }
            }
          },
`
And the Connection resource will look like:
`
{
        "type": "Microsoft.Web/connections",
        "apiVersion": "2016-06-01",
        "location": "[parameters('logicAppLocation')]",
        "name": "[parameters('LDAPAdapter_name')]",
        "properties": {
          "api": {
            "id": "[concat('/subscriptions/',subscription().subscriptionId,'/resourceGroups/',parameters('LDAPAdapter-ResourceGroup'),'/providers/Microsoft.Web/customApis/',parameters('LDAPAdapter_name'),'')]"
          },
          "displayName": "[parameters('LDAPAdapter_displayName')]",
          "parameterValues": {
            "authType": "[parameters('LDAPAdapter_authType')]",
            "gateway": {
              "id": "[concat('/subscriptions/',subscription().subscriptionId,'/resourceGroups/',parameters('LDAPAdapter_gatewayresourcegroup'),'/providers/Microsoft.Web/connectionGateways/',parameters('LDAPAdapter_gatewayname'))]"
            }
          }
        }
      }
`

Both of them has the id in a parameter as you reported, can you verify that you get this also or if we have something that makes the generation differently.

**Comment by mikaelsand** [comment]
@MLogdberg Are you talking to me?
If so, yes the thing I posted really happened. What kind of information do you feel you need?
I would be VERY happy to get this thing out of the "needs to be handled manually step".
Perhaps it is because I am using the SOAP passthru?

**Comment by MLogdberg** [comment]
@mikaelsand okey, then I'll take a look at it.

**Comment by erkoolio** [comment]
Hey!

When deploying arm template to a different subscription there maybe a need to look at resources from different subscriptions, is it possible to parametrize subscription id in logic app template when using GetARMTemplate script?
I have not checked if the data is available, but when deploying with hard coded value the deployment is successsful, so I'm guessing the data can be seen when creating template with the script.

**Comment by MLogdberg** [comment]
Yes it's possible to set subscriptionID as parameter as you mention now it's assuming it's in the same subscription. It will be some places it should be added. I'll take a look and come back to you

**Comment by erkoolio** [comment]
I added the subscriptionId as parameter, I'll do a pull request after I have merged the latest changes to my version.

**Comment by erkoolio** [comment]
I have the yesterdays latest version and I have parametrized subscriptionid, I did not do a pull request, I just edited master branch (I know, should have used pull request). I have also adjusted the tests to handle the subscription parametrization. (Meaning subscription.subscriptionid() => parameter('somename')
